third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -84,7 +101,7 @@ impl<'r> TransactionDataCache<'r> {
 
             let mut resources = BTreeMap::new();
             for (ty, (layout, gv)) in account_data_cache.data_map {
-                let op = match gv.into_effect() {
+                let op = match gv.into_effect_with_layout(layout) {
```

Comment:
wouldn't this whole block have better flow with if let Some(op) = .. {

}

instead of match & short circuit on None?

Reply:
No idea why it was like this, but let's use ` if let Some(op) = .. { ... }` instead.